packages/gatsby-transformer-remark/src/extend-node-type.js

```diff
@@ -133,6 +133,7 @@ module.exports = (
                     getNode,
                     reporter,
                     cache,
+                    ...rest,
```

Comment:
I wonder if doing something like:
```
cache: getCache(plugin.id)
```
to pass individual cache for subplugin woudln't make sense, so subplugins don't need to use `getCache` directly, and would automatically get same `cache` instance as in their `gatsby-node` API hooks? 

Questions:
 - Is there scenario where it is actually desirable that subplugin use parent plugin cache?
I can't think if reason for this of top of my head.
 - Will this break any subplugin?
Cache key generating functions ( https://github.com/gatsbyjs/gatsby/blob/6a8c59385d57b256d839fd861f3cc6be0e262cfc/packages/gatsby-transformer-remark/src/extend-node-type.js#L31-L50 ) are not exported, so shouldn't be considered as public API, so this shouldn't be considered as breaking change.

Reply:
Ultimately I would like to prep helper functions/objects in core, so parent plugin could pass prepped helpers, but this is out of scope for this PR I think